picocrt/machine/aarch64/crt0.c

```diff
@@ -32,24 +32,29 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
  * OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+#include <sys/cdefs.h>
 
 /* The size of the thread control block.
  * TLS relocations are generated relative to
  * a location this far *before* the first thread
  * variable (!)
+ * NB: The actual size before tp also includes padding
+ * to align up to the alignment of .tdata/.tbss.
  */
 #if __SIZE_WIDTH__ == 32
 #define TCB_SIZE	8
 #else
 #define TCB_SIZE	16
 #endif
+extern char __arm64_tls_tcb_offset;
+#define TP_OFFSET ((size_t)&__arm64_tls_tcb_offset)
```

Comment:
this needs to pick between the two tcb offset values -- the difference here is whether this is ilp32 ABI or lp64 ABI.

I assume the TCB contains two pointers; on ilp32, those are each 4 bytes, on lp64 they're 8 bytes.

Reply:
Should be fixed now, good catch!